src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+const service = require('../services/expenses.services');
+const { getById: getUserById } = require('../services/users.services');
+
+const getAllExpenses = (req, res) => {
+  const {
+    userId,
+    from,
+    to,
+  } = req.query;
+
+  const normalizedUrl = new URL('http://localhost:3000/' + req.url);
+
+  const categories = normalizedUrl.searchParams.getAll('categories');
+
+  let expenses = service.getAll();
+
+  if (userId) {
+    expenses = expenses.filter(expense => expense.userId === +userId);
+  }
+
+  if (from) {
+    expenses = expenses.filter(expense => {
+      return new Date(expense.spentAt).valueOf() >= new Date(from).valueOf();
+    });
+  }
+
+  if (to) {
+    expenses = expenses.filter(expense => {
+      return new Date(expense.spentAt).valueOf() <= new Date(to).valueOf();
+    });
+  }
+
+  if (categories.length) {
```

Comment:
`categories` is an array, so `if (categories)` will always be true

Reply:
With this one
You can use `console.log` to see why it will work correctly 
If we will not have categories in searchParams -> `categories===undefined`
If we have `categories===[array of categories]`